### PR TITLE
refactor!: Change configuration and device toml files to yaml

### PR DESCRIPTION
**Pull request by marcpfuller**
BREAKING CHANGE: Configuration and device files now use yaml

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) N/A
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) N/A
  <link to docs PR>

## Testing Instructions
1. start up edgex with no-secty
2. `make build`
3. run `./device-usb-camera -cp -o`
4. verify device-usb-camera  configuration loaded with no error

### CODE REVIEW

**Review by presatish — CHANGES_REQUESTED** [review]
There are document references with `toml` that needs to be changed to `yaml`.

**Review by lenny-goodell — CHANGES_REQUESTED** [review]
remove TOML module from Attribution.txt

**Comment by lenny-goodell** [comment]
@marcpfuller , you need to fix gofmt issue with driver.go and the provision watcher needs to be convert to YAML. Commit and PR tile and breaking message need to be update to include provision watcher

**Comment by marcpfuller** [comment]
> @marcpfuller , you need to fix gofmt issue with driver.go and the provision watcher needs to be convert to YAML. Commit and PR tile and breaking message need to be update to include provision watcher

@lenny-intel Provision watcher was already using yaml...

**Comment by lenny-goodell** [comment]
> @lenny-intel Provision watcher was already using yaml...

Yep, sorry, but the Commit and PR tile and breaking message still need to be updated to include provision watcher

**Comment by marcpfuller** [comment]
> There are document references with `toml` that needs to be changed to `yaml`.

are you referring to edgex docs of the document in this repo?

**Comment by presatish** [comment]
> > There are document references with `toml` that needs to be changed to `yaml`.
> 
> are you referring to edgex docs of the document in this repo?

Under here https://github.com/edgexfoundry/device-usb-camera/tree/main/docs. You can also refer https://github.com/edgexfoundry/device-onvif-camera/pull/281 which has similar changes.

File: docs/general-usage.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ For example:
     AutoStreaming = "false"
 ```
 <p align="left">
-      <i>Sample: Snippet from general.usb.camera.toml.example</i>
+      <i>Sample: Snippet from general.usb.camera.yaml.example</i>
```
Comment (lenny-goodell) [inline docs/general-usage.md:76]: As with ONVIF, the above in-line TOML example must be converted to YAML. Please make sure this is done for all in-line TOML.

**Review by lenny-goodell — CHANGES_REQUESTED** [review]

**Comment by presatish** [comment]
@marcpfuller Also with the latest main the provision watchers are not loading.

File: docs/general-usage.md (hunk 1 above)
Comment (marcpfuller) [reply docs/general-usage.md:76]: Yep I have already converted those example files to Yaml, both in the name of the file and the formatting of the file

Comment (lenny-goodell) [reply docs/general-usage.md:76]: @marcpfuller , not the files. The in-line TOML in the doc files.

**Comment by vyshali-chitikeshi** [comment]
attached is the  device usb error log with latest edgex main
[_edgex-device-usb-camera_logs (4).txt](https://github.com/edgexfoundry/device-usb-camera/files/11082558/_edgex-device-usb-camera_logs.4.txt)

File: cmd/res/devices/general.usb.camera.yaml.example [hunk 2]
```diff
@@ -0,0 +1,9 @@
+DeviceList:
+  Name: "example-camera"
+  ProfileName: "USB-Camera-General"
+  Description: "Example Camera"
+  Labels: [ "device-usb-camera-example", ]
+  DeviceList.Protocols:
+    DeviceList.Protocols.USB:
+      Path: "/dev/video0"
+      AutoStreaming: "false"
```
Comment (lenny-goodell) [inline cmd/res/devices/general.usb.camera.yaml.example]: The device files fields as need to be camelCase. This is due to the DTO's definition for YAML here specified camelCase.
https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/dtos/device.go#L16-L28
Note that the map keys like `USB` need to match constant used in the code.

File: cmd/res/devices/hp.w200.yaml.example [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Pre-define Devices
+# DeviceList:
+#   Name: "hp-w200-01"
+#   ProfileName: "HP-Webcam-w200"
+#   Description: "HP Webcam w200 - 01"
+#   Labels: [ "device-usb-camera-example" ]
+#   DeviceList.Protocols:
+#     DeviceList.Protocols.USB:
+#       Path: "/dev/video0"
+#       AutoStreaming: "false"
```
Comment (lenny-goodell) [inline cmd/res/devices/hp.w200.yaml.example]: Same changes here as to the other file

**Comment by vyshali-chitikeshi** [comment]
with the latest fixes, I don't see previous problem of provision watcher not loading. However it still fails to validate provision watcher profile. I see USB device discovered but not able to execute any api's, it looks like its not added to metadaa properly. attached is the usb service logs.
[_edgex-device-usb-camera_logs (6).txt](https://github.com/edgexfoundry/device-usb-camera/files/11084342/_edgex-device-usb-camera_logs.6.txt)

**Comment by marcpfuller** [comment]
> with the latest fixes, I don't see previous problem of provision watcher not loading. However it still fails to validate provision watcher profile. I see USB device discovered but not able to execute any api's, it looks like its not added to metadaa properly. attached is the usb service logs. [_edgex-device-usb-camera_logs (6).txt](https://github.com/edgexfoundry/device-usb-camera/files/11084342/_edgex-device-usb-camera_logs.6.txt)

hmm, @vyshali-chitikeshi  I am still looking into this error. I have checked dtos.ProvisionWatcher and I goind the ServiceName and ProfileName resides in DiscoveredDevice, which is correctly written in the generic_provision_watcher.yaml file. I am not sure why it is saying it required even though it is there

**Comment by marcpfuller** [comment]
@vyshali-chitikeshi Please re-review, I just needed update go mod to latest device-sdk and latest core-contracts

File: docs/general-usage.md [hunk 4]
```diff
@@ -62,18 +62,18 @@ The device's protocol properties contain:
...
-    AutoStreaming = "false"
+DeviceList:
+  Name: "hp-w200-01"
+  ProfileName: "USB-Camera-General"
+  Description: "HP Webcam w200 - 01"
+  Labels: [ "device-usb-camera-example" ]
+  Protocols
+    Protocols.USB:
+      Path: "/dev/video0"
+      AutoStreaming: "false"
```
Comment (lenny-goodell) [inline docs/general-usage.md]: These need to be camelCase same as in the file

File: docs/general-usage.md [hunk 5]
```diff
@@ -62,18 +62,18 @@ The device's protocol properties contain:
...
-    [DeviceList.Protocols.USB]
-    Path = "/dev/video0"
-    AutoStreaming = "false"
+DeviceList:
+  Name: "hp-w200-01"
+  ProfileName: "USB-Camera-General"
+  Description: "HP Webcam w200 - 01"
+  Labels: [ "device-usb-camera-example" ]
+  Protocols
+    Protocols.USB:
```
Comment (lenny-goodell) [inline docs/general-usage.md]:
```suggestion
    Protocols:
      USB:
```

**Review by lenny-goodell — CHANGES_REQUESTED** [review]

File: docs/general-usage.md (hunk 5 above)
Comment (marcpfuller) [reply docs/general-usage.md]: @lenny-intel shouldnt "Protocols" be "protocols". The field tag in dtos.Device for Protocols shows "protocol"

https://github.com/edgexfoundry/go-mod-core-contracts/blob/dacf12b8e974c6295a06680e4e8ee181d6834552/dtos/device.go#L26

**Comment by vyshali-chitikeshi** [comment]
Validation looks good in both secure and non-secure modes for USB camera. Discovered conencted usb cameras correctly and tested some rest-api's and it works correctly.

**Review by presatish — DISMISSED** [review]
LGTM

File: docs/general-usage.md (hunk 5 above)
Comment (lenny-goodell) [reply docs/general-usage.md]: Yes. :-)

File: cmd/res/devices/general.usb.camera.yaml.example [hunk 6]
```diff
@@ -0,0 +1,9 @@
+DeviceList:
+  name: "example-camera"
+  profileName: "USB-Camera-General"
+  description: "Example Camera"
+  labels: [ "device-usb-camera-example", ]
+  Protocols:
```
Comment (lenny-goodell) [inline cmd/res/devices/general.usb.camera.yaml.example]:
```suggestion
  protocols:
```

File: cmd/res/devices/hp.w200.yaml.example [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Pre-define Devices
+# DeviceList:
+#   name: "hp-w200-01"
+#   profileName: "HP-Webcam-w200"
+#   description: "HP Webcam w200 - 01"
+#   labels: [ "device-usb-camera-example" ]
+#   Protocols:
```
Comment (lenny-goodell) [inline cmd/res/devices/hp.w200.yaml.example]:
```suggestion
#   protocols:
```

File: docs/general-usage.md [hunk 8]
```diff
@@ -62,18 +62,18 @@ The device's protocol properties contain:
...
-  [DeviceList.Protocols]
-    [DeviceList.Protocols.USB]
-    Path = "/dev/video0"
-    AutoStreaming = "false"
+DeviceList:
+  name: "hp-w200-01"
+  profileName: "USB-Camera-General"
+  description: "HP Webcam w200 - 01"
+  labels: [ "device-usb-camera-example" ]
+  Protocols
```
Comment (lenny-goodell) [inline docs/general-usage.md]:
```suggestion
  protocols
```

File: docs/general-usage.md [hunk 9]
```diff
@@ -62,18 +62,18 @@ The device's protocol properties contain:
...
+    USB:
+      path: "/dev/video0"
+      autoStreaming: "false"
```
Comment (lenny-goodell) [inline docs/general-usage.md]: These are map keys and should stay how they were in the TOML
```suggestion
      Path: "/dev/video0"
      AutoStreaming: "false"
```

File: cmd/res/devices/hp.w200.yaml.example [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+#     USB:
+#       path: "/dev/video0"
+#       autoStreaming: "false"
```
Comment (lenny-goodell) [inline cmd/res/devices/hp.w200.yaml.example]: These are map keys and should stay how they were in the TOML so they match the constants used in the code
```suggestion
#       Path: "/dev/video0"
#       AutoStreaming: "false"
```

File: cmd/res/devices/general.usb.camera.yaml.example [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+    USB:
+      path: "/dev/video0"
+      autoStreaming: "false"
```
Comment (lenny-goodell) [inline cmd/res/devices/general.usb.camera.yaml.example]: These are map keys and should stay how they were in the TOML so they match the constants used in the code

```suggestion
      Path: "/dev/video0"
      AutoStreaming: "false"
```

**Review by lenny-goodell — CHANGES_REQUESTED** [review]

**Review by lenny-goodell — APPROVED** [review]
LGTM

**Pull request merged**